src/Queue/Queue.php

```diff
@@ -0,0 +1,140 @@
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license.
+ */
+
+namespace ZfrEbWorker\Queue;
+
+use Aws\Sqs\SqsClient;
+use ZfrEbWorker\Exception\RuntimeException;
+use ZfrEbWorker\Message\MessageInterface;
+
+/**
+ * @author Michaël Gallego
+ */
+class Queue implements QueueInterface
+{
+    /**
+     * @var array[]
+     */
+    private $messages = [];
+
+    /**
+     * @var SqsClient
+     */
+    private $sqsClient;
+
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var string
+     */
+    private $url;
+
+    /**
+     * @param SqsClient $sqsClient
+     * @param string    $name
+     * @param string    $url
+     */
+    public function __construct(SqsClient $sqsClient, string $name, string $url = '')
+    {
+        $this->sqsClient = $sqsClient;
+        $this->name      = $name;
+        $this->url       = $url;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function setQueueUrl(string $url)
+    {
+        $this->url = $url;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function push(MessageInterface $message, array $options = [])
+    {
+        $this->messages[] = [
+            'options' => $options,
+            'body'    => [
+                'name'    => $message->getName(),
+                'payload' => $message->getPayload()
+            ]
+        ];
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function flush(bool $async = false)
+    {
+        if (null === $this->url) {
+            throw new RuntimeException(sprintf(
+                'Queue "%s" is not mapped to an actual SQS queue URL. Did you make sure you have specified the
+                 queue into the "zfr_eb_worker" config?',
+                $this->name
+            ));
+        }
+
+        $this->doFlush($async);
+
+        // We reset the messages so that we make sure we don't duplicate message by calling flush multiple times
+        $this->messages = [];
+    }
+
+    /**
+     * Flush a queue
+     *
+     * @param  bool $async
+     * @return void
+     */
+    private function doFlush(bool $async)
+    {
+        // SQS only supports batch of 10, so we need to splice like that
+
+        while (!empty($this->messages)) {
+            $messagesToPush = array_splice($this->messages, 0, 10);
+
+            $parameters = [
+                'QueueUrl' => $this->url,
+                'Entries'  => []
+            ];
+
+            foreach ($messagesToPush as $key => $message) {
+                $messageParameters = [
+                    'Id'           => $key, // Identifier of the message in the batch
+                    'MessageBody'  => json_encode($message['body']),
+                    'DelaySeconds' => $message['options']['delay_seconds'] ?? null
+                ];
+
+                $parameters['Entries'][] = array_filter($messageParameters, function ($value) {
+                    return $value !== null;
```

Comment:
I don't remmeber why I did that. This code is extracted from SlmQueue, but I remember that I had to do this to make it work, so I'm not going to touch it :D.